### ISSUE TITLE
Pyright python suggestions not work properly

### ISSUE THREAD

**Issue by giovanniOfficioso** [zed-industries/zed#17530]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

The suggestion in pyright don't work properly, both with base and local environment. I'm on miniconda.
No suggestion and a problem, that not exists really, highlighted by pyright.

I'm working with virtual environment using conda, but the issue still in the base environment.
Is possible to fix it?
Here is also my setting:
```json
{
  "vim_mode": true,
  "telemetry": {
    "diagnostics": false,
    "metrics": false
  },
  "ui_font_size": 16,
  "buffer_font_size": 13,
  "buffer_font_family": "MesloLGS NF",
  "buffer_font_weight": 400,
  "ui_font_features": {
    "calt": true
  },
  "buffer_font_features": {
    "ss02": true
  },
  "theme": {
    "mode": "system",
    "light": "One Dark Vivid",
    "dark": "One Dark Vivid"
  },
  "lsp": {
    "ruff": {
      "initialization_options": {
        "settings": {
          "lineLength": 80,
          "lint": {
            "extendSelect": ["I"]
          }
        }
      }
    }
  },
  "languages": {
    "Python": {
      "format_on_save": "on",
      "formatter": [
        {
          "code_actions": {
            "source.organizeImports.ruff": true,
            "source.fixAll.ruff": true
          }
        }
      ],
      "tab_size": 4,
      "language_servers": ["ruff", "pyright"]
    }
  },
  "current_line_highlight": "all",
  "show_whitespaces": "all",
  "indent_guides": {
    "enabled": true,
    "line_width": 1,
    "active_line_width": 1,
    "coloring": "indent_aware",
    "background_coloring": "indent_aware"
  },
  "show_wrap_guides": true,
  "terminal": {
    "font_family": "MesloLGS NF"
  },
  "cursor_blink": true
}
```

### Environment

Zed: v0.152.1 (Zed Preview)
OS: macOS 13.6.7
Memory: 32 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

When I write the code here is as it appears in zed: 
![Screenshot 2024-09-07 at 12 34 30](https://github.com/user-attachments/assets/f2e8b9b2-cf59-4914-9cef-5c37ac01008d)

Here is the same windows in fleet editor, with both linter and suggestions which work properly:
![Screenshot 2024-09-07 at 12 35 18](https://github.com/user-attachments/assets/31ddbadd-3f26-4e9f-9e94-831249e3dedc)

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by mak448a** [comment]
You need to add it to your settings. `"show_completions_on_input": true,`.

**Comment by giovanniOfficioso** [comment]
Hi man, also with your suggestion, the issue still. Pyright doesn't provide suggestion on. based on what I code, but only on standard libraries that I installed in the virtual environment.
![Screenshot 2024-09-08 at 10 55 06](https://github.com/user-attachments/assets/20beebc6-cc53-4860-9490-6dcb254d6280)
As you can see I have no suggestion for the "df" object, but in other tools it there is. Not only fleet, also in viscide, as you can see from the following screen.
![Screenshot 2024-09-08 at 10 57 02](https://github.com/user-attachments/assets/479b252a-4f30-4393-9105-27d0eccb8442)
So the issue still both for suggestion and for the inexistent highlighted error.

**Comment by sdhmh** [comment]
Did you create your `pyrightconfig.json`?

You will have to link it to your conda env

For example, with `~/miniconda3/` as the base directory where miniconda is installed you have to create a pyrightconfig.json like this:
```json
{
  "venv": "env_name",
  "venvPath": "~/miniconda3/envs/"
}
```
You will have to change the path if you have installed it globally (like `/opt/miniconda3/envs/`)

If you are using anaconda the directory might be `anaconda`. (Double check it)

**Comment by mak448a** [comment]
Did you add your venv to your global config?

**Comment by Moshyfawn** [comment]
I see your issue from 11 August reporting a similar problem. Based on the configuration you provided, it appears that you are reporting the same issue. Is this correct?

**Comment by giovanniOfficioso** [comment]
The issues are different because if I install ruff, pyright doesn’t work because it doesn’t provide any suggestion. So they seem incompatible. If I install only pyright, the suggestions are very poor if they are compared with the other ones provided in other tools and also the code checker doesn’t work in the right way because it highlights errors which really don’t exists. The main problems appears to be pyright here and it would be awesome if we could have alternatives such as Jedi in order to have a better coding experience.

**Comment by sdhmh** [comment]
yes I had the exact same issue when I installed ruff as an extension in zed.

But using these options in my `settings.json` I was able to get both ruff (formatting) and pyright (linting and suggestions) working together. 

(Note: Ruff is still not installed as an extension)

```json
"languages": {
    "Python": {
      "format_on_save": {
        "external": {
          "command": "bash",
          "arguments": [
            "-c",
            "ruff check --select I --fix --stdin-filename {buffer_path} | ruff format --stdin-filename {buffer_path}"
          ]
        }
      }
    }
  },
```

This uses Ruff as (just) the formatter and Pyright as the language server.

**Comment by brendan-corrigan** [comment]
> The issues are different because if I install ruff, pyright doesn’t work because it doesn’t provide any suggestion. So they seem incompatible. If I install only pyright, the suggestions are very poor if they are compared with the other ones provided in other tools and also the code checker doesn’t work in the right way because it highlights errors which really don’t exists. The main problems appears to be pyright here and it would be awesome if we could have alternatives such as Jedi in order to have a better coding experience.

This issue has some weird behaviour, I believe it has to do with the order they are loaded in.

I managed to get around this by changing the order when specifying language servers.

Going from: 
`"language_servers": ["ruff", "pyright"]`
To:
`"language_servers": ["pyright", "ruff"]`

**Comment by giovanniOfficioso** [comment]
After changing nothing work, no suggestion and worse code checking

**Comment by wxitcode** [comment]
1. languages -> Python `"language_servers": ["pyright", "ruff"]`
2. lsp -> pyright  `"pythonPath": ".venv/bin/python"`

``` json
{
  ...
  "languages": {
      "Python": {
        **"language_servers": ["pyright", "ruff"],**
        "format_on_save": "on",
        "formatter": [
          {
            "code_actions": {
              "source.organizeImports.ruff": true,
              "source.fixAll.ruff": true
            }
          },
          {
            "language_server": {
              "name": "ruff"
            }
          }
        ]
      }
    },
  "lsp": {
      "pyright": {
        "settings": {
          "python.analysis": {
            "diagnosticMode": "workspace",
            "typeCheckingMode": "strict"
          },
          "python": {
            **"pythonPath": ".venv/bin/python"**
          }
        }
      }
  ...
}
```

**Comment by liquidcarbon** [comment]
Taking a suggestion for taming Pylance, this is how you keep your sanity in Zed.

```
{
  "lsp": {
    "pyright": {
      "settings": {
        "python.analysis": {
          "ignore": ["*"]
        }
      }
    }
  },
  "languages": {
    "Python": {
      "language_servers": ["ruff", "pyright"],
      "format_on_save": "on",
      "formatter": [
        {
          "code_actions": {
            "source.organizeImports.ruff": true,
            "source.fixAll.ruff": true
          }
        }
      ]
    }
  }
}
```

Ruff is on (here highlighting the f-string without placeholders).
Pyright stays on and does the type hints and autocompletes, but does not complain about anything.

![image](https://github.com/user-attachments/assets/137bfa5f-0ddc-43ec-8ea8-ddeb56249dd0)


In VSCode:
```
{
    "python.analysis.ignore": [ "*" ],
    "[python]": {
        "editor.formatOnSave": true,
        "editor.defaultFormatter": "charliermarsh.ruff"
    }   
}
```
